tensorflow/core/kernels/map_kernels.h

```diff
@@ -176,6 +173,43 @@ class TensorMapHasKey : public OpKernel {
   }
 };
 
+class TensorMapStackKeys : public OpKernel {
+ public:
+  explicit TensorMapStackKeys(OpKernelConstruction* c) : OpKernel(c) {
+    OP_REQUIRES_OK(c, c->GetAttr("key_dtype", &key_dtype_));
+  }
+  ~TensorMapStackKeys() override {}
+
+  void Compute(OpKernelContext* c) override {
+    const TensorMap* m = nullptr;
+    OP_REQUIRES_OK(c, GetInputMap(c, 0, &m));
+    
+    OP_REQUIRES(c, m->size() != 0,
```

Comment:
Why not just return an empty result of shape [0, key_shape] instead of failing?

Reply:
We do not know the `key_shape` here because the map is empty. One option would be to pass in the expected key_shape but that could be implemented in user code with something like. 

```
size = TensorMapSize(m)
if size == 0:
  return tf.zeros([0] + key_shape)
else:
  return TensorMapStackKeys(m)
```